src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs

```diff
@@ -194,16 +218,70 @@ private static void SetElement(XmlDocument document, XmlNode node, string name,
                 UpdateVersionFiles(versionProps, globalJson, itemToUpdate);
             }
 
+            // Combine the two sets of dependencies. If an asset is present in the itemsToUpdate,
+            // prefer that one over the old dependencies
+            Dictionary<String, HashSet<string>> itemsToUpdateLocations = GetAssetLocationMapping(itemsToUpdate);
+
+            if (oldDependencies != null)
+            {
+                foreach (DependencyDetail dependency in oldDependencies)
+                {
+                    if (!itemsToUpdateLocations.ContainsKey(dependency.Name) && dependency.Locations != null)
+                    {
+                        itemsToUpdateLocations.Add(dependency.Name, new HashSet<string>(dependency.Locations));
+                    }
+                }
+            }
+
+            // At this point we only care about the Maestro managed locations for the assets. 
+            // Flatten the dictionary into a set that has all the managed feeds
+            HashSet<string> managedFeeds = FlattenLocations(itemsToUpdateLocations, IsManagedFeed);
+
+            var updatedNugetConfig = UpdatePackageSources(nugetConfig, managedFeeds);
+
             var fileContainer = new GitFileContentContainer
             {
                 GlobalJson = new GitFile(VersionFiles.GlobalJson, globalJson),
                 VersionDetailsXml = new GitFile(VersionFiles.VersionDetailsXml, versionDetails),
-                VersionProps = new GitFile(VersionFiles.VersionProps, versionProps)
+                VersionProps = new GitFile(VersionFiles.VersionProps, versionProps),
+                NugetConfig = new GitFile(VersionFiles.NugetConfig, updatedNugetConfig)
             };
 
             return fileContainer;
         }
 
+        private bool IsManagedFeed(string feed)
```

Comment:
>Managed [](start = 23, length = 7)

What does managed entails?

Reply:
I'm using the term managed to mean that maestro manages that feed. Tooooootally open to suggestions.

Reply:
It makes sense and I thought that would be the case. Probably just add a comment somewhere explaining what does managed mean

---
In reply to: [295942517](https://github.com/dotnet/arcade-services/pull/477#discussion_r295942517) [](ancestors = 295942517)

Reply:
Will do

Reply:
IsMaestroManagedFeed perhaps a better name?

Reply:
Sure, I think I like that one